src/map/utils/synthutils.cpp

```diff
@@ -572,23 +499,132 @@ int32 doSynthSkillUp(CCharEntity* PChar)
                     }
                 }
 
+                // Cap skill gain if character hits the current cap
                 if((skillAmount + charSkill) > maxSkill)
                 {
                     skillAmount = maxSkill - charSkill;
                 }
 
-                PChar->RealSkills.skill[skillID] += skillAmount;
-                PChar->pushPacket(new CMessageBasicPacket(PChar, PChar, skillID, skillAmount, 38));
-
-                if((charSkill/10) < (charSkill + skillAmount)/10)
+                if (charSkill > 700) // The cap from which you are limited to 400 points
                 {
-                    PChar->WorkingSkills.skill[skillID] += 0x20;
+                    uint16 skillCumulation = 0;
+                    uint8 skillHighest    = 0;
+
+                    // we are gonna set which is the OTHER highest skill over lvl 70 and the amount of points we have of the 400 to allocate
+                    if (PChar->RealSkills.skill[49] > 700) // Craft 1
+                    {
+                        skillCumulation += (PChar->RealSkills.skill[49] - 700);
```

Comment:
Perhaps all these 8 skill checks could be replaced with one fairly good for loop!
```c++
for(int i = SKILL_WOODWORKING; i <= SKILL_COOKING; i++)
{
    if (PChar->RealSkills.skill[i] > 700) // Craft 2
    {
        skillCumulation += (PChar->RealSkills.skill[i] - 700);
        if (skillID != i && PChar->RealSkills.skill[i] > skillHighest)
            skillHighest = i;
    }
}
```

Reply:
I'm also unsure if it's intentional, but wouldn't `skillCumulation` be reduced for every skill below 70? (ie 500 - 700 would be -200 to skillCumulation).

Reply:
Thank you mate. Done